### DISCUSSION TITLE
[Routing] LookupError: no template for url_hash=''

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#346] [Q&A] [answered]
I am trying to make the upgrade to v2.1 but seem to be having troubles moving to the new template version of routing.

[Here is a clone link](https://anvil.works/build#clone:KNM7WRAY22YB7Z7K=LOFG3SD747JY46APGMMQW5UM)

Basic premise is that I have an app built without templates and would like to roll over the code to a template version.
I have changed my start up form which used `open_form()` after running some login functions.

I have now moved this to a start up module which calls `routing.launch()` but it raises an error that I presume means the home form is needing a template?

`LookupError: no template for url_hash=''  # error raised from routing.launch()` 

I intend on using the templates but only for limited access views, the rest of the time I wanted to get away with just using the default template (ie MainForm).

What am I missing here? It feels like I am doing something dumb but cannot find it for the life of me!

**Comment by s-cork** [comment, marked as answer]
You just need to import MainForm into startup module + any other routes and templates

**Reply by rhurlbatt** [reply]
> If you are using a Startup Module or a Startup Form all the TemplateForms and RouteForms must be imported otherwise they will not be registered by the routing module.

I see this now... 🤦‍♂️ 

Atleast I was right about one thing ... I was doing something dumb 😃 

Thank you Stu!!
